main.go

```diff
@@ -221,194 +211,64 @@ func main() {
 }
 
 func validateRepo(c *cli.Context) {
+	actions.ChartsScriptOptionsFile = ChartsScriptOptionsFile
 	if LocalMode && RemoteMode {
 		logrus.Fatalf("cannot specify both local and remote validation")
 	}
 
-	CurrentPackage = "" // Validate always runs on all packages
-	chartsScriptOptions := parseScriptOptions()
-
-	logrus.Infof("Checking if Git is clean")
-	_, _, status := getGitInfo()
-	if !status.IsClean() {
-		logrus.Warnf("Git is not clean:\n%s", status)
-		logrus.Fatal("Repository must be clean to run validation")
+	if LocalMode {
+		actions.ValidateLocal()
+		return
 	}
 
 	if RemoteMode {
-		logrus.Infof("Running remote validation only, skipping generating charts locally")
-	} else {
-		logrus.Infof("Generating charts")
-		generateCharts(c)
-
-		logrus.Infof("Checking if Git is clean after generating charts")
-		_, _, status = getGitInfo()
-		if !status.IsClean() {
-			logrus.Warnf("Generated charts produced the following changes in Git.\n%s", status)
-			logrus.Fatalf("Please commit these changes and run validation again.")
-		}
-		logrus.Infof("Successfully validated that current charts and assets are up to date.")
-	}
-
-	if chartsScriptOptions.ValidateOptions != nil {
-		if LocalMode {
-			logrus.Infof("Running local validation only, skipping pulling upstream")
-		} else {
-			repoRoot := getRepoRoot()
-			repoFs := filesystem.GetFilesystem(repoRoot)
-			releaseOptions, err := options.LoadReleaseOptionsFromFile(repoFs, "release.yaml")
-			if err != nil {
-				logrus.Fatalf("Unable to unmarshall release.yaml: %s", err)
-			}
-			u := chartsScriptOptions.ValidateOptions.UpstreamOptions
-			branch := chartsScriptOptions.ValidateOptions.Branch
-			logrus.Infof("Performing upstream validation against repository %s at branch %s", u.URL, branch)
-			compareGeneratedAssetsResponse, err := validate.CompareGeneratedAssets(repoFs, u, branch, releaseOptions)
-			if err != nil {
-				logrus.Fatal(err)
-			}
-			if !compareGeneratedAssetsResponse.PassedValidation() {
-				// Output charts that have been modified
-				compareGeneratedAssetsResponse.LogDiscrepancies()
-				logrus.Infof("Dumping release.yaml tracking changes that have been introduced")
-				if err := compareGeneratedAssetsResponse.DumpReleaseYaml(repoFs); err != nil {
-					logrus.Errorf("Unable to dump newly generated release.yaml: %s", err)
-				}
-				logrus.Infof("Updating index.yaml")
-				if err := helm.CreateOrUpdateHelmIndex(repoFs); err != nil {
-					logrus.Fatal(err)
-				}
-				logrus.Fatalf("Validation against upstream repository %s at branch %s failed.", u.URL, branch)
-			}
-		}
-	}
-
-	logrus.Info("Zipping charts to ensure that contents of assets, charts, and index.yaml are in sync.")
-	zipCharts(c)
-
-	logrus.Info("Doing a final check to ensure Git is clean")
-	_, _, status = getGitInfo()
-	if !status.IsClean() {
-		logrus.Warnf("Git is not clean:\n%s", status)
-		logrus.Fatal("Repository must be clean to pass validation")
+		actions.ValidateRemote()
+		return
```

Comment:
there is already a Validate function that we should call in the actions repo, otherwise it should be removed from there.